### PR TITLE
Security Group Exists Check

### LINKED ISSUE #13: aws-security-group-exists
```
- aws-security-group-exists : true
```

### LINKED ISSUE #14: aws-ec2-instance-uses-security-group
```
- aws-ec2-instance-uses-security-group : sg-1234567
- aws-ec2-instance-uses-security-group : BaseLinux
- aws-ec2-instance-uses-security-group :
  - sg-1234567
  - BaseLinux
```

### PR DESCRIPTION
**Pull request by tgwilliams63**
Checking to see if security groups defined in terraform exist in AWS. 

resolve #13 and resolve #14

### CODE REVIEW

**Comment by tgwilliams63** [comment]
resolve #15 as well

**Comment by tgwilliams63** [comment]
resolve #17 as well

**Comment by tgwilliams63** [comment]
resolve #16

**Comment by tgwilliams63** [comment]
resolve #18

File: lib/rules/aws-ec2-instance-uses-security-group.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+'use strict';
+const debug = require('debug')('tfrules/aws-ec2-instance-uses-security-group');
```
Comment (clearly) [inline lib/rules/aws-ec2-instance-uses-security-group.js]: Change to aws-ec2-uses-security-group

File: lib/rules/aws-security-group-allow-inbound.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+'use strict';
+const co = require('co')
+const debug = require('debug')('tfrules/aws-security-group-allow-inbound');
```
Comment (clearly) [inline lib/rules/aws-security-group-allow-inbound.js]: allow -> allows

File: lib/rules/aws-security-group-allow-inbound.spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+'use strict';
+const AWS = require('aws-stub');
+const chai = require('chai');
+const rule = require('./aws-security-group-allow-inbound');
```
Comment (clearly) [inline lib/rules/aws-security-group-allow-inbound.spec.js]: allow -> allows

File: lib/rules/aws-security-group-allow-outbound.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+'use strict';
+const co = require('co')
+const debug = require('debug')('tfrules/aws-security-group-allow-outbound');
```
Comment (clearly) [inline lib/rules/aws-security-group-allow-outbound.js]: allow -> allows

File: lib/rules/aws-security-group-does-not-allow-inbound.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
+'use strict';
+const co = require('co')
+const debug = require('debug')('tfrules/aws-security-group-does-not-allow-inbound');
+const _ = require('lodash');
+var ip = require('ip')
+const inRange = require('in-range');
+const util = require('util')
```
Comment (clearly) [inline lib/rules/aws-security-group-does-not-allow-inbound.js]: 'util' required but not used?

File: lib/rules/aws-security-group-allow-inbound.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+const _ = require('lodash');
+var ip = require('ip')
+const inRange = require('in-range');
```
Comment (clearly) [inline lib/rules/aws-security-group-allow-inbound.js]: Would this be useful?
https://www.npmjs.com/package/ip-cidr

File: lib/rules/aws-security-group-allow-outbound.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+  }
+};
+
+AWSSecurityGroupAllowOutbound.paths = {
+  awsSecurityGroup : 'aws_security_group'
+};
+
+let cidrComparison = function (config, instance, name) {
+  let currCidr = ip.cidrSubnet(config.cidr)
+  _.map(instance.cidr_blocks, block => {
```
Comment (clearly) [inline lib/rules/aws-security-group-allow-outbound.js]: Not sure what this is returning?

File: terraform.tfrules [hunk 8]
```diff
@@ -17,3 +17,18 @@ rules :
...
+  - sg-24a9b85d
+- aws-security-group-does-not-allow-inbound :
+    cidr : '204.4.6.2/32'
+    port : '9032-9034'
+- aws-security-group-does-not-allow-outbound :
+    cidr : '204.4.6.2/32'
+    port : '9032-9034'
+- aws-security-group-allow-inbound : 
+    cidr : '127.0.0.1/32'
+    port : '80'
```
Comment (clearly) [inline terraform.tfrules]: For single numbers a number or string should be allowed

File: lib/rules/aws-ec2-instance-uses-security-group.js (hunk 1 above)
Comment (tgwilliams63) [inline lib/rules/aws-ec2-instance-uses-security-group.js]: Would you like me to change the file name for the .js and .spec.js files as well?

**Pull request merged**